### PR TITLE
FaIR: Check allowed emissions

### LINKED ISSUE #32: FaIR1 adapter: problem with unit conversion
When running the FaIR1 adapter with a scenario that has e.g. 'Emissions|CO2' in addition to 'Emissions|CO2|MAGICC AFOLU' i think there is a problem as it tries to convert the unit with whatever was the previous unit.

I think it needs to handle the Exception differently, maybe always with `continue` or reset `fair_unit`?
If an Exception occurs, `fair_unit` is compared against the previous value (unless in the list of vars below).

https://github.com/openscm/openscm-runner/blob/master/src/openscm_runner/adapters/fair_adapter/_scmdf_to_emissions.py#L215

```python
for var_df in scmrun.groupby("variable"):
        variable = var_df.get_unique_meta("variable", no_duplicates=True)
        in_unit = var_df.get_unique_meta("unit", no_duplicates=True)
        try:
            fair_col, fair_unit, context = _get_fair_col_unit_context(variable)
        except AssertionError:
            if variable.endswith(
                (
                    "|HFC152a",
                    "|HFC236fa",
                    "|HFC365mfc",
                    "|NF3",
                    "|C3F8",
                    "|C4F10",
                    "|C5F12",
                    "|C7F16",
                    "|C8F18",
                    "|cC4F8",
                    "|SO2F2",
                    "|CH2Cl2",
                    "|CHCl3",
                )
            ):
                continue
    
    if in_unit != fair_unit:
        var_df_fair_unit = var_df.convert_unit(fair_unit, context=context)
    else:
        var_df_fair_unit = var_df
```

Thoughts, @chrisroadmap?

### PR DESCRIPTION
**Pull request by rgieseke**
Replaces #33 

Closes #32 

- [x] Tests added
- [ ] Documentation added
- [ ] Example added (in the documentation, to an existing notebook, or in a new notebook)
- [x] Description in ``CHANGELOG.rst`` added (single line such as: ``(`#XX <https://github.com/openscm/openscm-runner/pull/XX>`_) Added feature which does something``)

### CODE REVIEW

**Comment by rgieseke** [comment]
Can you give this a look @chrisroadmap?

https://github.com/openscm/openscm-runner/pull/37/commits/c9af8d042e8d067482b2fe49c4e4f0e262861498

I'm not sure what the assert was trying to guard against, but i think it's safer now by just throwing an error if there is a problem.

**Comment by chrisroadmap** [comment]
That looks like it will do the job.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -27,6 +27,7 @@ Fixed
 ~~~~~
 
 - (`#36 <https://github.com/openscm/openscm-runner/pull/36>`_) Hotfix CI after pandas 1.1.5 broke pylint
+- (`#37 <https://github.com/openscm/openscm-runner/pull/33>`_) Stop FaIR trying to run aggregated CO2 emissions
```
Comment (znicholls) [inline CHANGELOG.rst:30]:
```suggestion
- (`#37 <https://github.com/openscm/openscm-runner/pull/33>`_) Stop FaIR trying to run with total CO2 emissions
```

Trying to think of a clearer way to explain the change

File: tests/unit/test_fair1x_utils.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import numpy.testing as npt
+
+from openscm_runner.adapters.fair_adapter._scmdf_to_emissions import scmdf_to_emissions
+
+
+# All emissions that are not covered by FaIR (see
+# EMISSIONS_SPECIES_UNITS_CONTEXT) should be ignored, e.g.
+# 'Emissions|CO2' (which is summarizing 'MAGICC Fossil
```
Comment (znicholls) [inline tests/unit/test_fair1x_utils.py:8]:
```suggestion
# 'Emissions|CO2' (which is the sum of any sub-categories)
```

and line below can be deleted

**Review by znicholls — APPROVED** [review]
lgtm, minor suggested tweaks

File: CHANGELOG.rst (hunk 1 above)
Comment (rgieseke) [reply CHANGELOG.rst]: Changed it to 

- (`#37 <https://github.com/openscm/openscm-runner/pull/33>`_) Ensure FaIR ignores emissions input in scenarios not handled by FaIR, e.g. total CO2

Hope that's better!

**Pull request merged**